src/actions/ui.js

```diff
@@ -1,21 +1,23 @@
 // @flow
 const constants = require("../constants");
-const { getSource, getFileSearchState } = require("../selectors");
+const { getSource, getSearchFieldState } = require("../selectors");
 import type { ThunkArgs } from "./types";
 
-function toggleFileSearch() {
+function toggleSearchVisibility(field: string, toggleValue?: boolean) {
   return ({ dispatch, getState }: ThunkArgs) => {
-    dispatch({
-      type: constants.SET_FILE_SEARCH,
-      searchOn: !getFileSearchState(getState())
-    });
-  };
-}
-
-function closeFileSearch() {
-  return {
-    type: constants.SET_FILE_SEARCH,
-    searchOn: false
+    if (toggleValue != null) {
```

Comment:
This was done  to remove the need for a separate action creator for closing the search field.